node-backend-starter-code/client.js

```diff
@@ -0,0 +1,391 @@
+	var searchingFor = searchInput.value;
+
+	//Disable the button and change the text so the user knows we are searching
+	searchBtn.text = "Searching...";
+	searchInput.disabled = searchBtn.disabled = true;
+
+	//time to make our ajax request
+	var xmlhttpRequest = new XMLHttpRequest();
+	xmlhttpRequest.onreadystatechange = function(){
+		if ( xmlhttpRequest.readyState === 4 ){
+			var parsed = JSON.parse(xmlhttpRequest.responseText);
+			if ( xmlhttpRequest.status === 200 && parsed.Response !== "False"){
+				populateMovies( parsed.Search, false );
+			}
+			else {
+				displayError("Sorry but there was an error with your request, please try again.");
+			}
+			//reenable our search button
+			searchBtn.text = "Go!";
+			searchInput.disabled = searchBtn.disabled = false;
+		}
+	};
+
+	xmlhttpRequest.open("GET", encodeURI(omdbSearchUrl + searchingFor), true);
+	xmlhttpRequest.send(null);
+}
+//handle search tab being select
+function onNavSearchClick() {
+	navSearch.className = "active";
+	navHearts.className = ( uuid ) ? "" : "hidden";
+	searchContainer.className = "";
+	heartsContainer.className = "hidden";
+
+	clearMovies();
+	//stop the link being processed
+	event.preventDefault();
+}
+//handle nav tab being select
+function onNavHeartsClick() {
+	navSearch.className = "";
+	navHearts.className = "active";
+	searchContainer.className = "hidden";
+	heartsContainer.className = "";
+
+	loadHearts();
+	//stop the link being processed
+	event.preventDefault();
+}
+
+//nice function found on stackoverflow for getting the url parameters - is that cheating?
+function getParameterByName(name) {
+    name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+    var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
+        results = regex.exec(location.search);
+    return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
+}
+
+//all initialisation happens in our page load event
+window.onload = function(){ 
+
+	//check if we recieved a uid for the user
+	setUUID( getParameterByName("uuid") );
+
+	//cache our commonly referenced elements
+	searchContainer = document.getElementById("searchContainer");
+	heartsContainer = document.getElementById("heartsContainer");
+	searchInput = document.getElementById("searchInput");
+	searchBtn = document.getElementById("searchBtn");
+	navSearch = document.getElementById("navSearch");
+	navHearts = document.getElementById("navHearts");
+	errorMessage = document.getElementById("errorMessage");
+	moviesContainer = document.getElementById("movies");
+	shareLink = document.getElementById("shareLink");
+
+	//add an event listener for tab selection and form processing
+	navSearch.addEventListener("click", onNavSearchClick, false);
+	navHearts.addEventListener("click", onNavHeartsClick, false);
+	searchBtn.addEventListener("click", onSearchClick, false);
+	//also allow an enter key press to trigger the search
+	searchInput.addEventListener("keydown", function(e){ 
```

Comment:
How about wrapping in a `<form>` element with a `submit` event? Any issues with search input type and/or submit button/input type?